### PR TITLE
fix(billing): allow non-trial users to remove payment methods

### PR DESCRIPTION
**Pull request by baktun14**
## Why

Non-trial users were unable to remove their payment methods. The backend already correctly allows removal for non-trial users, but the frontend unconditionally hid the "Remove" action for default/only payment methods regardless of trial status.

Closes CON-40

## What

- Thread `isTrialing` from `useWallet()` and `isAutoReloadEnabled` from `useWalletSettingsQuery()` through the PaymentMethods component chain (Container → View → Row)
- Relax frontend removal restrictions for non-trial users while keeping them for trial users
- Block removal of the default payment method when auto-reload is enabled to prevent edge cases
- Add comprehensive tests for all trial/non-trial/auto-reload scenarios

## Summary by CodeRabbit

* **New Features**
  * Payment method actions now respect trial status and auto-reload settings — action menu, "Set as default" and "Remove" options appear or hide appropriately per account state.

* **Tests**
  * Added comprehensive tests covering propagation of trial and auto-reload flags and verifying action visibility and behavior across combinations of states.

### CODE REVIEW

File: apps/deploy-web/src/components/billing-usage/PaymentMethodsContainer/PaymentMethodsContainer.tsx [hunk 1]
```diff
@@ -21,6 +24,9 @@ export const PaymentMethodsContainer: React.FC<PaymentMethodsContainerProps> = (
     refetch: refetchPaymentMethods,
     isRefetching: isRefetchingPaymentMethods
   } = d.usePaymentMethodsQuery();
+  const { isTrialing } = d.useWallet();
+  const { data: walletSettings } = d.useWalletSettingsQuery();
+  const isAutoReloadEnabled = walletSettings?.autoReloadEnabled ?? false;
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/components/billing-usage/PaymentMethodsContainer/PaymentMethodsContainer.tsx]: Fixed in 9b616b6ce — now defaults to `isWalletSettingsLoading` instead of `false`, preventing transient removal before settings resolve.

File: apps/deploy-web/src/components/billing-usage/PaymentMethodsView/PaymentMethodsRow.tsx [hunk 2]
```diff
@@ -96,41 +100,48 @@ export const PaymentMethodsRow: React.FC<PaymentMethodsRowProps> = ({
     closeMenu();
   }, [onRemovePaymentMethod, paymentMethod.id]);
 
+  const canSetAsDefault = !paymentMethod.isDefault && hasOtherPaymentMethods;
+  const isDefaultBlockedByAutoReload = paymentMethod.isDefault && isAutoReloadEnabled;
+  const canRemove = !isDefaultBlockedByAutoReload && (!isTrialing || (hasOtherPaymentMethods && !paymentMethod.isDefault));
+  const showActions = canSetAsDefault || canRemove;
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/components/billing-usage/PaymentMethodsView/PaymentMethodsRow.tsx:106]: Fixed in 9b616b6ce — simplified `canRemove` to `!isDefaultBlockedByAutoReload && !isTrialing`, blocking trial users entirely.

File: apps/deploy-web/src/components/billing-usage/PaymentMethodsView/PaymentMethodsView.spec.tsx [hunk 3]
```diff
@@ -17,12 +17,19 @@ const mockUseTheme = vi.fn(() => ({
...
-const MockPaymentMethodsRow = ({ paymentMethod, onSetPaymentMethodAsDefault, onRemovePaymentMethod, hasOtherPaymentMethods }: any) => (
-  <tr data-testid={`payment-method-row-${paymentMethod.id}`}>
+const MockPaymentMethodsRow = ({
+  paymentMethod,
+  onSetPaymentMethodAsDefault,
+  onRemovePaymentMethod,
+  hasOtherPaymentMethods,
+  isTrialing,
+  isAutoReloadEnabled
+}: any) => (
```
[the remark this answers is not in the document]
Comment (baktun14) [reply apps/deploy-web/src/components/billing-usage/PaymentMethodsView/PaymentMethodsView.spec.tsx]: Fixed in 9b616b6ce — replaced `any` with `PaymentMethodsRowProps`.

**Review by stalniy — APPROVED** [review]
